cargo-dist/src/backend/ci/github.rs

```diff
@@ -253,11 +258,18 @@ fn install_dist_for_github_runner<'a>(
     runner: GithubRunner,
     install_sh: &'a str,
     install_ps1: &'a str,
-) -> &'a str {
-    if runner == GITHUB_LINUX_RUNNER || runner == GITHUB_MACOS_RUNNER {
-        install_sh
+    targets: &[&String],
+) -> String {
+    if runner == GITHUB_LINUX_RUNNER && targets.contains(&&"x86_64-unknown-linux-musl".to_owned()) {
+        // Necessary to ensure we have the proper toolchain
+        format!(
+            r#"rustup target add x86_64-unknown-linux-musl
+{install_sh}"#
+        )
```

Comment:
cargo-dist already has logic for calling rustup internally, you should be generalizing that instead https://github.com/axodotdev/cargo-dist/blob/7544b23b5fa938f0bb6e2b46b1dece9a79ed157b/cargo-dist/src/tasks.rs#L1875-L1889